lib/src/ima.rs

```diff
@@ -47,10 +48,19 @@ fn xattrs_to_map(v: &glib::Variant) -> BTreeMap<Vec<u8>, Vec<u8>> {
     map
 }
 
-/// Reserialize a map to GVariant of type `a(ayay)`
-fn xattrmap_serialize(map: &BTreeMap<Vec<u8>, Vec<u8>>) -> glib::Variant {
-    let map: Vec<_> = map.iter().collect();
-    variant_utils::new_variant_a_ayay(&map)
+/// Create a new GVariant of type a(ayay).  This is used by OSTree's extended attributes.
+fn new_variant_a_ayay<'a, T: 'a + AsRef<[u8]>>(
```

Comment:
Not gonna lie after I wrote this I felt like I'd just inscribed a new page in a spellbook.